### PR TITLE
Upgrade extension to JupyterLab 3.0

### PR DESCRIPTION
**Pull request by philippjfr**

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -40,8 +41,8 @@ export class NBWidgetExtension implements INBWidgetExtension {
   }
 }
 
-import {name, version} from "./metadata"
-import {BokehModel, BokehView} from "./widgets"
+declare const name: string
+declare const version: string
```
Comment (philippjfr) [inline src/plugin.ts:45]: @mattpap Need your help with this. It didn't like the `metadata` import.

File: setup.py [hunk 2]
```diff
@@ -1,68 +1,93 @@
...
-    include_package_data=True,
-    data_files=[
-        ("share/jupyter/nbextensions/jupyter_bokeh", [
-            "jupyter_bokeh/nbextension/static/extension.js",
-            "jupyter_bokeh/nbextension/static/index.js",
-            "jupyter_bokeh/nbextension/static/index.js.map",
-        ]),
-        ("etc/jupyter/nbconfig/notebook.d", [
-            "jupyter_bokeh.json",
-        ]),
```
Comment (philippjfr) [inline setup.py:63]: @mattpap We'll somehow have to combine the new JupyterLab compilation stuff with the old nbextension stuff.

File: package.json [hunk 3]
```diff
@@ -12,11 +13,22 @@
...
+    "clean:all": "jlpm run clean:lib && jlpm run clean:labextension",
+    "clean:labextension": "rimraf pyviz_comms/labextension",
+    "clean:lib": "rimraf lib tsconfig.tsbuildinfo",
+    "eslint": "eslint . --ext .ts,.tsx --fix",
+    "eslint:check": "eslint . --ext .ts,.tsx",
+    "install:extension": "jupyter labextension develop --overwrite .",
+    "prepare": "jlpm run clean && jlpm run build:prod",
+    "watch": "run-p watch:src watch:labextension",
+    "watch:labextension": "jupyter labextension watch .",
+    "watch:src": "tsc -w"
```
Comment (philippjfr) [inline package.json:31]: @mattpap How much of the old scripts was custom? Do we have to keep a `build:nbextension` script or similar?

**Comment by philippjfr** [comment]
@mattpap @bryevdv This is a start, haven't tested it yet but it does compile. Hope to get back to it next week. Sorting out the nbextension vs. labextension building and whether those can be unified is another project.

**Comment by bryevdv** [comment]
superseded by #124

**Pull request closed without merging**